waterbutler/providers/rushfiles/provider.py

```diff
@@ -88,7 +88,24 @@ async def revalidate_path(self,
                               base: WaterButlerPath,
                               name: str,
                               folder: bool=None) -> WaterButlerPath:
-        raise NotImplementedError # Or user super if appropriate
+        response = await self.make_request(
+            'GET',
+            self.build_url(str(self.share['id']), 'virtualfiles', base.identifier, 'children'),
+            expects=(200, 404,),
+            throws=exceptions.MetadataError,
+        )
+        if response.status == 404:
+            raise exceptions.NotFoundError(name)
+        res = await response.json()
+        child_id, index = self._search_inter_id(res, name)
+
+        if res['Data'][index]['IsFile'] == folder:
+            raise exceptions.NotFoundError(name)
+
+        if child_id is None:
+            raise exceptions.NotFoundError(name)
```

Comment:
`revalidate_path` should behave more like `validate_path` and not through when path is not found. Only set the id to None. Checking for file existence is done using metadata/exists (MetadataError(404) error thrown from there is caught)

https://github.com/tsukaeru/RDM-waterbutler/blob/develop/waterbutler/server/api/v1/provider/create.py#L58-L85